[18.0][MIG] fieldservice_stage_validation: Migration to 18.0

fsmw opened the pull request:
Migration of fieldservice_stage_validation module to Odoo 18.0.

## Summary
- ✅ **Proper migration using git format-patch method** - preserves complete git history from 17.0 branch
- ✅ Updated module version to 18.0.1.0.0  
- ✅ **Fixed translation issue** - use rec.env._ for proper translation context
- ✅ Applied all pre-commit fixes and formatting
- ✅ Follows OCA migration guidelines exactly

## Module Description
This module adds field validation to FSM stages, allowing administrators to configure required fields that must be set before moving to specific stages in the field service workflow.

## Migration Details
- **Method**: Used git format-patch to preserve complete git history (not manual copy)
- **Code Changes**: Updated translation calls to use proper rec.env._ context
- **Quality**: All pre-commit linting and formatting applied
- **Testing**: Ready for Runboat CI testing environment

## Technical Changes
- Fixed: "no translation language detected" warning in validate_utils.py
- Updated: Manifest version format to 18.0.1.0.0
- Preserved: Full commit history from 17.0 branch

Closes part of migration checklist in OCA/field-service#1220

brian10048 committed:
[ADD] fieldservice_stage_validation

brian10048 committed:
[FIX] fieldservice_stage_validation

Travis and miscellaneous errors

OCA-git-bot committed:
fieldservice_stage_validation 12.0.0.2.0

dreispt committed:
Added translation using Weblate (Portuguese (Portugal))

dreispt committed:
Translated using Weblate (Portuguese (Portugal))

Currently translated at 100.0% (10 of 10 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice_stage_validation
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice_stage_validation/pt_PT/

brian10048 committed:
[IMP] fieldservice_stage_validation: black, isort, prettier

brian10048 committed:
[MIG] fieldservice_stage_validation: Migration to 13.0

brian10048 committed:
[IMP] fieldservice_stage_validation: Add test

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (10 of 10 strings)

Translation: field-service-13.0/field-service-13.0-fieldservice_stage_validation
Translate-URL: https://translation.odoo-community.org/projects/field-service-13-0/field-service-13-0-fieldservice_stage_validation/es_AR/

sanrav committed:
Added translation using Weblate (French (France))

sanrav committed:
Translated using Weblate (French (France))

Currently translated at 100.0% (10 of 10 strings)

Translation: field-service-13.0/field-service-13.0-fieldservice_stage_validation
Translate-URL: https://translation.odoo-community.org/projects/field-service-13-0/field-service-13-0-fieldservice_stage_validation/fr_FR/

tate11 committed:
Added translation using Weblate (Turkish)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (10 of 10 strings)

Translation: field-service-13.0/field-service-13.0-fieldservice_stage_validation
Translate-URL: https://translation.odoo-community.org/projects/field-service-13-0/field-service-13-0-fieldservice_stage_validation/it/

yankinmax committed:
[IMP] fieldservice_stage_validation: black, isort, prettier

yankinmax committed:
[MIG] fieldservice_stage_validation: Migration to 14.0

ShwetaSerpentCS committed:
[FIX] pre-commit.

ShwetaSerpentCS committed:
Update pre-commit.yml

ShwetaSerpentCS committed:
[FIX] fieldservice_stage_validation: test

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (10 of 10 strings)

Translation: field-service-15.0/field-service-15.0-fieldservice_stage_validation
Translate-URL: https://translation.odoo-community.org/projects/field-service-15-0/field-service-15-0-fieldservice_stage_validation/es_AR/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (10 of 10 strings)

Translation: field-service-15.0/field-service-15.0-fieldservice_stage_validation
Translate-URL: https://translation.odoo-community.org/projects/field-service-15-0/field-service-15-0-fieldservice_stage_validation/it/

macagua committed:
Added translation using Weblate (Spanish)

macagua committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (10 of 10 strings)

Translation: field-service-15.0/field-service-15.0-fieldservice_stage_validation
Translate-URL: https://translation.odoo-community.org/projects/field-service-15-0/field-service-15-0-fieldservice_stage_validation/es/

Commit:
[MIG] fieldservice_stage_validation: Migration to 16.0

Commit:
[UPD] Update fieldservice_stage_validation.pot

OCA-git-bot committed:
[UPD] README.rst

adrianojprado committed:
Added translation using Weblate (Portuguese (Brazil))

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (10 of 10 strings)

Translation: field-service-16.0/field-service-16.0-fieldservice_stage_validation
Translate-URL: https://translation.odoo-community.org/projects/field-service-16-0/field-service-16-0-fieldservice_stage_validation/pt_BR/

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: field-service-16.0/field-service-16.0-fieldservice_stage_validation
Translate-URL: https://translation.odoo-community.org/projects/field-service-16-0/field-service-16-0-fieldservice_stage_validation/

Commit:
[IMP] fieldservice_stage_validation: pre-commit auto fixes

Commit:
[MIG] fieldservice_stage_validation: Migration to 17.0

Commit:
[UPD] Update fieldservice_stage_validation.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (10 of 10 strings)

Translation: field-service-17.0/field-service-17.0-fieldservice_stage_validation
Translate-URL: https://translation.odoo-community.org/projects/field-service-17-0/field-service-17-0-fieldservice_stage_validation/it/

ivantodorovich committed:
[FIX] fieldservice_stage_validation: missing rights to read validate_field_ids

OCA-git-bot committed:
[BOT] post-merge updates

ivantodorovich commented:
Hello!

Thanks for the contribution.

It seems, though, the module commit history is not preserved. 
Did you follow the steps mentioned here: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-18.0#technical-method-to-migrate-a-module-from-170-to-180-branch ?

Alternatively, there are these two libraries that make the process a bit smoother:
- https://github.com/oca/odoo-module-migrator
- https://github.com/oca/oca-port

fsmw committed:
[IMP] fieldservice_stage_validation: pre-commit auto fixes

fsmw committed:
[FIX] fieldservice_stage_validation: Use rec.env._ for proper translation context

fsmw committed:
[FIX] fieldservice_stage_validation: Apply pre-commit fixes

fsmw committed:
[FIX] fieldservice_repair: Fix test setup to ensure stock.quant is created before equipment

The test setup was creating equipment objects before the corresponding
stock.quant records. Since the equipment's current_stock_location_id is
a computed field that depends on stock.quant records, the field couldn't
find the location when equipment was created first.

By reordering the setup to create stock.quant before equipment, the
computed field can properly resolve the current inventory location,
fixing the test_action_view_repairs_multiple test failure.

fsmw commented:
> Hello!
> 
> Thanks for the contribution.
> 
> It seems, though, the module commit history is not preserved. Did you follow the steps mentioned here: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-18.0#technical-method-to-migrate-a-module-from-170-to-180-branch ?
> 
> Alternatively, there are these two libraries that make the process a bit smoother:
> 
>     * https://github.com/oca/odoo-module-migrator
> 
>     * https://github.com/oca/oca-port

I added previous history, since is my fist contribution am still learning the process.

ivantodorovich commented:
There are some files added in one of your commits, and then removed in a following one:
<img width="3486" height="1786" alt="image" src="https://github.com/user-attachments/assets/8881c9f4-1c31-443b-a00e-ccc329604ef7" />


Please clean that up.
Also, maybe squash your commits into a single "Migration to 18.0" one, or at least keep only one for "pre-commit autofixes"
